nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py

```diff
@@ -132,55 +295,216 @@ def __init__(self):
         self.unlock_right_gripper()
 
     def cmd_right_gripper(self, goal, blocking=True):
-        if self.gripper_lock["right"]:
-            self.right_gripper.send_goal(goal)
-            if blocking:
-                return self.right_gripper.wait_for_result()
-            return True  # TODO: action_server.get_result()
-        return False
+        if not self.gripper_lock["right"]:
+            return False
 
-    def cmd_left_gripper(self, goal, blocking=True):
-        if self.gripper_lock["left"]:
-            self.left_gripper.send_goal(goal)
-            if blocking:
-                return self.left_gripper.wait_for_result()
-            return True  # TODO: action_server.get_result()
-        return False
+        self.right_gripper.send_goal(goal)
+        self.right_gripper.wait_for_result()
+        return self.right_gripper.get_result().status == "Success"
+
+    def cmd_left_gripper(self, goal):
+        if not self.gripper_lock["left"]:
+            return False
+
+        self.left_gripper.send_goal(goal)
+        self.left_gripper.wait_for_result()
+        return self.left_gripper.get_result().status == "Success"
 
     def lock_right_gripper(self):
-        self.gripper_lock["right"] = True
+        self.gripper_lock["right"] = False
```

Comment:
Are these booleans correct? It feels like if we are locking the gripper it should be True and vice versa

Reply:
i agree its logically backwards, but it should still work. you can bring this up in the V1 PR